app/models/SelfAssessmentUserType.scala

```diff
@@ -16,15 +16,55 @@
 
 package models
 
+import play.api.libs.json.{JsDefined, JsError, JsResult, JsString, JsSuccess, JsValue, Json, Reads, Writes}
 import uk.gov.hmrc.domain.SaUtr
 
 sealed trait SelfAssessmentUserType
+
 sealed trait SelfAssessmentUser extends SelfAssessmentUserType {
   def saUtr: SaUtr
+}
+
+object SelfAssessmentUserType {
+  val cacheId = "SelfAssessmentUser"
+
+  implicit val writes = new Writes[SelfAssessmentUserType] {
+    override def writes(o: SelfAssessmentUserType): JsValue = o match {
+      case ActivatedOnlineFilerSelfAssessmentUser(utr) =>
+        Json.obj("_type" -> JsString("ActivatedOnlineFilerSelfAssessmentUser"), "utr" -> JsString(utr.toString))
```

Comment:
Better to just ask the object for it's name to ensure it's always the same, if someone were to change the class name it would be a one place fix: 

`ActivatedOnlineFilerSelfAssessmentUser.toString`